accepted/future-releases/class-modifiers/feature-specification.md

```diff
@@ -933,8 +931,139 @@ as a mixin. If the class defines a generative constructor or extends anything
+are:
+
+- Did you say both `abstract` and `sealed`?  Drop `abstract`; it’s redundant.
+  Now there's only 95 possibilities.
+
+- Did you say both `interface` and `final`?  Drop `interface`; it’s redundant.
+  Now there's only 71 possibilities.
+
+- Did you say both `base` and `final`?  Drop `base`; it’s redundant.  Now
+  there's only 59 possibilities.
+
+- Did you say both `interface` and `base`?  Say `final` instead.  Now there's
+  only 47 possibilities.
+
+- Did you say neither `mixin` nor `class`?  You have to pick one or the other or
+  both.  The fixup can probably safely assume you mean `class`.  (Exception: if
+  you just said `interface` and no other keywords, you probably mean `abstract
+  class`).  Now there's only 36 possibilities.
+
+- Did you say both `sealed` and `final`?  Drop `final`; it’s redundant.  Now
+  there's only 33 possibilities.
+
+- Did you say both `sealed` and `base`?  Drop `base`; it’s redundant.  Now
+  there's only 30 possibilities.
+
+- Did you say both `sealed` and `interface`?  Drop `interface`; it’s redundant.
+  Now there's only 27 possibilities.
+
+- Did you say both `mixin` and `class`, as well as one of the following
+  keywords: `sealed`, `interface`, or `final`?  Drop `class` and replace
+  `extends M` with `with M` wherever it appears in your library.  Now there's
+  only 22 possibilities.
+
+- Did you say both `abstract` and `mixin`, but not `class`?  Drop `abstract`;
+  it’s redundant.  Now we are down to the 18 permitted possibilities.
+
+If we take this sort of approach, then users who don't love reading
+documentation will be able to just experimentally string together combinations
+of the keywords we've made available to them, and the errors and fixups will
+guide them to something valid, and then they can play around and see the effect.
+
+### Introducing the feature to users
+
+If we assume that most users will have access to the IDE fixups noted above, it
+suggests that a nice way to introduce the feature to folks would be to gloss
+over what combinations are redundant or contradictory, and just tell them in
+plain English what each keyword does.  Users who love reading documentation can
+read further and find out which combinations are prohibited; users who don't can
+just try them out, and the IDE will train them which combinations are valid over
+time.  So the core of the feature becomes explainable in just seven lines, three
+of which are just restatements of things the user was already familiar with.
+Something like:
+
+- `sealed` means "this type has a known set of direct subtypes, so switching on
+  it will require the switch to be exhaustive".
+
+- `abstract` means "this type can't be constructed directly", but you already
+  knew that.  It's only included in the list to help clarify that `abstract` is
+  one of the seven keywords users should try combining together at the top of
+  your declaration.
+
+- `interface` means "this type can't be extended from outside this library".
```

Comment:
I think it is clearer to present this in terms of what the user *can* do.  That is `interface` means "this type can only be implemented (not extended) outside of its defining library".  Likewise for the `base` below.  It seems small, but I think it's much clearer - if you tell people what they *can't do*, then they need to subtract from the set of all possible things in order to find out what they *can do*, which is usually what they want to know.